templates/baseService.handlebars

```diff
@@ -1,10 +1,12 @@
 /* tslint:disable */
-import { {{configurationClass}} } from './{{configurationFile}}';
+import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
+import { {{configurationClass}} } from './{{configurationFile}}';
 
 /**
  * Base class for services
  */
+@Injectable()
```

Comment:
Why decorating the base service with `@Injectable`?
Each concrete subclass has the annotation.

Reply:
This apparently is a new requirement for base classes in Angular 9. See: https://angular.io/guide/deprecations#undecorated-base-classes

Technically, since the base class doesn't use any of the listed "Angular features", you would think this requirement doesn't apply to this service base class. However, during automated migration to angular 9 using `ng update @angular/core@9`, if you have ng-openapi-gen generated services, it will add this to the generated services. I suspect "dependency injection constructor arguments" count as an "angular feature" but isn't listed in the above document.

Reply:
I'm not sure if DI is considered an Angular feature requiring the @Injectable decorator as peer the docs, but having it doesn't hurt, so I chose to side on the err of caution by adding it..

Reply:
there is also this migration which adds missing `@Injectable()` https://angular.io/guide/migration-injectable, but again i'm not sure if it strictly applies in this instance